test/test_geos_types.jl

```diff
@@ -350,4 +350,23 @@ testValidTypeDims(multipoly::LibGEOS.MultiPolygon) =
         LibGEOS.destroyGeom(geomcollect_ctx)
     end
 
-end
\ No newline at end of file
+end
+
+@testset "Multi threading" begin
+    function f91(n)
+        # adapted from https://github.com/JuliaGeo/LibGEOS.jl/issues/91#issuecomment-1267732709
+        contexts = [LibGEOS.GEOSContext() for i=1:Threads.nthreads()]
+        p = [[[-1.,-1],[+1,-1],[+1,+1],[-1,+1],[-1,-1]]]
+        Threads.@threads :static for i=1:n
+            ctx = contexts[Threads.threadid()]
+            g1 = LibGEOS.Polygon(p, ctx)
+            g2 = LibGEOS.Polygon(p, ctx)
```

Comment:
```julia 
LibGEOS.intersects(geo1 #=global context=#, geo2 #=global context=#, ctx #= local context =#)
```

Reply:
@nraynaud is that supported by the C-library? Was this on purpose or an oversight to use differenc contexts?

Reply:
To clarify doing this operation in C code is it known to you whether that is safe?

Reply:
Yes, we now know it's unsafe in C.

Reply:
So I think it might be a good idea to carry context around with objects and also add an assertion 
in every function that contexts of all involved objects are equal. What do you think?

Reply:
Ok I will add assertions in a follow up PR to keep this one as small as possible.